Controller/DiaporamaFrontController.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/*************************************************************************************/
+/*      This file is part of the "Diaporamas" Thelia 2 module.                       */
+/*                                                                                   */
+/*      Copyright (c) OpenStudio                                                     */
+/*      email : dev@thelia.net                                                       */
+/*      web : http://www.thelia.net                                                  */
+/*                                                                                   */
+/*      For the full copyright and license information, please view the LICENSE.txt  */
+/*      file that was distributed with this source code.                             */
+/*************************************************************************************/
+
+
+namespace Diaporamas\Controller;
+
+use Diaporamas\Event\DiaporamaEvent;
+use Diaporamas\Event\DiaporamaEvents;
+use Thelia\Controller\Front\BaseFrontController;
+use Thelia\Core\HttpFoundation\Response;
+
+class DiaporamaFrontController extends BaseFrontController
+{
+    public function getDiaporamaHtmlAction($shortcode)
+    {
+        $width = $this->getRequest()->query->get('width');
+        $height = $this->getRequest()->query->get('height');
+        $event = new DiaporamaEvent();
+        $event->__set('image_width', $width);
```

Comment:
Why not declaring those properties in your class and then create getters and setters methods ? What did you try to do ?